diag_app/static/js/script.js

```diff
@@ -126,25 +126,55 @@ function charRemainingText() {
 // return: none
 //////////////////////////////////////////////////////////////////////////
 function postProblem() {
-    var bike =  $("#probModel option:selected").val();
-    var sys =  $("#probSystem option:selected").val();
-    var text = $("#probText").val();
-    var user = $("#userId").val();
-    var header = $("#probTitle").val();
+    var modal = $('[data-remodal-id=askModal]').remodal();
+    var bike =  $("#probModel option:selected");
+    var sys =  $("#probSystem option:selected");
+    var text = $("#probText");
+    var user = $("#userId");
+    var header = $("#probTitle");
     var context = {
-        system: sys,
-        description: text,
-        tech: user,
-        model: bike,
-        title: header,
+        system: sys.val(),
+        description: text.val(),
+        tech: user.val(),
+        model: bike.val(),
+        title: header.val(),
     }
     $.ajax({
         url: '/api/post-problems/',
         type: 'POST',
         data: context,
-    }).done(function(results){
+        success: function (response) {
+          modal.close();
+          alert("Problem saved successfully!");
+        },
+        error: function (xhr, ajaxOptions, thrownError) {
+          $("p.error-message").empty();
```

Comment:
My only concern here is what if the error that is thrown does not conform to one of these three options? Also, you're using == when you should be using ===.

Reply:
Just made a commit with the changes you wanted. Hope that solves your problem!